starter/pages/_app.js

```diff
@@ -1,26 +1,29 @@
-import App from "next/app";
-import Head from "next/head";
-import ErrorPage from "next/error";
-import { useRouter } from "next/router";
-import { DefaultSeo } from "next-seo";
-import { getStrapiMedia } from "utils/media";
-import { getStrapiURL, getGlobalData } from "utils/api";
-import Layout from "@/components/layout";
-import "@/styles/index.css";
+import App from "next/app"
+import Head from "next/head"
+import ErrorPage from "next/error"
+import { useRouter } from "next/router"
+import { DefaultSeo } from "next-seo"
+import { getStrapiMedia } from "utils/media"
+import { getPageData, getGlobalData } from "utils/api"
+import Layout from "@/components/layout"
+import { parseCookies } from "utils/parse-cookies"
+import "@/styles/index.css"
 
 const MyApp = ({ Component, pageProps }) => {
   // Prevent Next bug when it tries to render the [[...slug]] route
-  const router = useRouter();
+  const router = useRouter()
   if (router.asPath === "/[[...slug]]") {
-    return null;
+    return null
   }
 
   // Extract the data we need
-  const { global } = pageProps;
+  const { global, initialLocale } = pageProps
```

Comment:
When accessing `/fr` on first load, `pageProps` is null, hits the 404 then then redirects to the correct to `/fr`.  `pageProps` is not null in `getInitialProps`.  This only happens in development, with a production build everything works as expected.